flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/tier/disk/DiskIOScheduler.java

```diff
@@ -236,18 +236,18 @@ private int readBuffersFromFile() {
     }
 
     private List<ScheduledSubpartitionReader> sortScheduledReaders() {
+        List<ScheduledSubpartitionReader> scheduledReaders;
         synchronized (lock) {
             if (isReleased) {
                 return new ArrayList<>();
             }
-            List<ScheduledSubpartitionReader> scheduledReaders = new ArrayList<>();
-            for (ScheduledSubpartitionReader reader : allScheduledReaders.values()) {
-                reader.prepareForScheduling();
-                scheduledReaders.add(reader);
-            }
-            Collections.sort(scheduledReaders);
-            return scheduledReaders;
+            scheduledReaders = new ArrayList<>(allScheduledReaders.values());
+        }
+        for (ScheduledSubpartitionReader reader : scheduledReaders) {
```

Comment:
We need a test to validate this is fixed, even though such tests may not be easy to write.

Reply:
I've add a test in `DiskIOScheduler`.